napari_aicssegmentation/core/layer_reader.py

```diff
@@ -0,0 +1,28 @@
+from typing import List
+from aicsimageio import AICSImage
+from napari.layers import Layer
+from napari_aicssegmentation.model.channel import Channel
+
+
+class LayerReader:
+    """
+    Reader / Helper class to extract information out of Napari Layers
+    """
+
+    def get_channels(self, layer: Layer) -> List[Channel]:
+        """
+        Get the list of image channels from a layer
+        TODO this is a workaround for now and we just guess the Channel dimension based on its
+             location for most ome tiffs
+        TODO use aicsimageio to read image from the source file path and get channel names
+             once Napari exposes Image layer source (next release)
+        """
+        if layer is None:
+            return None
+
+        img = AICSImage(layer.data)  # gives us a 6D image
+
+        channels = list()
+        for index in range(img.shape[3]):
```

Comment:
I tested this with an ometiff file that Jianxu gave me to test with a while back, and the plugin interpreted the Z dimension as the channels :/  Maybe this is ok since it's just temporary and we'll have a more robust solution in the near future??? 

I wonder if it'd be worth putting a length check for the list as a really-imperfect-but-should-work-most-of-the-time temporary solution. For example if `img.shape[3]` length is more than, say, 20, then use `img.shape[4]` or whatever? 😬 

![image](https://user-images.githubusercontent.com/12690133/114083940-7874bc80-9864-11eb-9b1c-582e1116b742.png)

Reply:
Can you send me that file? I need more files to test with. The ones i've tested so far work with this code and are also ome tiff, I'm curious as to why these would be formatted differently.

Reply:
Works now with the little heuristic I added.